tests/validations/test_time_spines.py

```diff
@@ -182,3 +188,89 @@ def test_dimension_granularity_smaller_than_time_spine() -> None:  # noqa: D
         "configuring a time spine at or below the smallest time dimension granularity is recommended"
         in issues.warnings[0].message
     )
+
+
+def test_time_spines_with_invalid_names() -> None:  # noqa: D
+    semantic_manifest = PydanticSemanticManifest(
+        semantic_models=[
+            semantic_model_with_guaranteed_meta(
+                name="semantic_model",
+                measures=[
+                    PydanticMeasure(
+                        name="sum_measure", agg=AggregationType.SUM, agg_time_dimension="dim", create_metric=True
+                    )
+                ],
+                dimensions=[
+                    PydanticDimension(
+                        name="dim",
+                        type=DimensionType.TIME,
+                        type_params=PydanticDimensionTypeParams(time_granularity=TimeGranularity.SECOND),
+                    )
+                ],
+                entities=[PydanticEntity(name="entity", type=EntityType.PRIMARY)],
+            ),
+        ],
+        metrics=[
+            PydanticMetric(
+                name="metric",
+                description=None,
+                type=MetricType.SIMPLE,
+                type_params=PydanticMetricTypeParams(measure=PydanticMetricInputMeasure(name="sum_measure")),
+            ),
+        ],
+        project_configuration=PydanticProjectConfiguration(
+            time_spine_table_configurations=[],
+            time_spines=[
+                PydanticTimeSpine(
+                    node_relation=PydanticNodeRelation(alias="time_spine", schema_name="my_fav_schema"),
+                    primary_column=PydanticTimeSpinePrimaryColumn(name="ds", time_granularity=TimeGranularity.MONTH),
+                    custom_granularities=[
+                        PydanticTimeSpineCustomGranularityColumn(name="retail_year"),
+                    ],
+                ),
+                PydanticTimeSpine(
+                    node_relation=PydanticNodeRelation(alias="time_spine2", schema_name="my_fav_schema"),
+                    primary_column=PydanticTimeSpinePrimaryColumn(name="ds", time_granularity=TimeGranularity.DAY),
+                    custom_granularities=[
+                        PydanticTimeSpineCustomGranularityColumn(name="retail_year"),
+                        PydanticTimeSpineCustomGranularityColumn(name="quarter"),  # todo
```

Comment:
Is something blocking this validation? Seems like its not in this PR

Reply:
whoops I just forgot to remove the comment! It is validated here!

Reply:
it's the "names cannot match reserved time granularity keywords" one

Reply:
fixed it!

Reply:
ooo didn't realize it was already in the code inside `check_valid_name` and that was done prior to this PR